application/src/main/java/org/togetherjava/tjbot/commands/moderation/ReportCommand.java

```diff
@@ -0,0 +1,224 @@
+
+        configModMailChannelPattern = config.getModMailChannelPattern();
+    }
+
+    private Cache<Long, Instant> createCooldownCache() {
+        return Caffeine.newBuilder()
+            .maximumSize(1_000)
+            .expireAfterAccess(COOLDOWN_DURATION_VALUE, TimeUnit.of(COOLDOWN_DURATION_UNIT))
+            .build();
+    }
+
+    @Override
+    public void onMessageContext(MessageContextInteractionEvent event) {
+        long userID = event.getUser().getIdLong();
+        String reportedMessageTimestamp = event.getTarget().getTimeCreated().toInstant().toString();
+
+        if (handleIsOnCooldown(event)) {
+            return;
+        }
+        authorToLastReportInvocation.put(userID, Instant.now());
+
+        String reportedMessage = event.getTarget().getContentRaw();
+        String reportedMessageID = event.getTarget().getId();
+        String reportedMessageChannel = event.getTarget().getChannel().getId();
+        String reportedAuthorName = event.getTarget().getAuthor().getName();
+        String reportedAuthorAvatarURL = event.getTarget().getAuthor().getAvatarUrl();
+        String reportedAuthorID = event.getTarget().getAuthor().getId();
+
+        TextInput body = TextInput
+            .create(REPORT_REASON_INPUT_ID, "Anonymous report to the moderators",
+                    TextInputStyle.PARAGRAPH)
+            .setPlaceholder("What is wrong with the message, why do you want to report it?")
+            .setRequiredRange(3, 200)
+            .build();
+
+        String reportModalComponentID = generateComponentId(reportedMessage, reportedMessageID,
+                reportedMessageChannel, reportedMessageTimestamp, reportedAuthorName,
+                reportedAuthorAvatarURL, reportedAuthorID);
+        Modal reportModal = Modal.create(reportModalComponentID, "Report this to a moderator")
+            .addActionRow(body)
+            .build();
+
+        event.replyModal(reportModal).queue();
+    }
+
+    private boolean handleIsOnCooldown(MessageContextInteractionEvent event) {
+        if (!isAuthorOnCooldown(event.getUser().getIdLong())) {
+            return false;
+        }
+        event
+            .reply("You can only report a message once per %s minutes."
+                .formatted(COOLDOWN_DURATION_VALUE))
+            .setEphemeral(true)
+            .queue();
+        return true;
+    }
+
+    private boolean isAuthorOnCooldown(long userId) {
+        return Optional.ofNullable(authorToLastReportInvocation.getIfPresent(userId))
+            .map(sinceCommandInvoked -> sinceCommandInvoked.plus(COOLDOWN_DURATION_VALUE,
+                    COOLDOWN_DURATION_UNIT))
+            .filter(Instant.now()::isBefore)
+            .isPresent();
+    }
+
+    @Override
+    public void onModalSubmitted(ModalInteractionEvent event, List<String> args) {
+        Optional<TextChannel> modMailAuditLog = handleRequireModMailChannel(event);
+
+        if (modMailAuditLog.isEmpty()) {
+            return;
+        }
```

Comment:
send some message at to the user at least??

Reply:
Done in handleRequireModMailChannel